micro-rdk/src/common/moisture_sensor.rs

```diff
@@ -37,7 +36,7 @@ impl Readings for MoistureSensor {
 
 impl SensorT<f64> for MoistureSensor {
     fn get_readings(&self) -> anyhow::Result<TypedReadingsResult<f64>> {
-        let reading = self.analog.borrow_mut().read()?;
+        let reading = self.analog.lock().unwrap().read()?;
```

Comment:
Also curious why you need the explicit lock/unwrap here.

Reply:
here I needed it because the receiver is `&self` and not `&mut self`, which forces me to rely on interior mutability (which `MutexGuard` has)